### PR TITLE
Remove SUMO accounts endpoints and functionality.

### PR DESCRIPTION
**Pull request by akatsoulas**

### CODE REVIEW

**Comment by akatsoulas** [comment]
@safwanrahman Could you have a look at this?

File: kitsune/questions/events.py [hunk 1]
```diff
@@ -86,8 +84,7 @@ def _make_mail(locale, user, context):
             is_asker = asker_id == user.id
             extra_params = {}
             if is_asker:
-                auth_str = get_auth_str(user)
-                extra_params['auth'] = auth_str
+                extra_params['auth'] = reverse('users.fxa_authentication_init')
```
Comment (LeoMcA) [inline kitsune/questions/events.py]: Is this some django magic which redirects users to the url in the `?auth=` parameter if they're not logged in, because I can't see where we're doing this ourselves?

File: kitsune/settings.py [hunk 2]
```diff
@@ -806,13 +796,10 @@ def JINJA_CONFIG():
 
 #
 # Sessions
-SESSION_COOKIE_AGE = config('SESSION_COOKIE_AGE', default=4 * 7 * 24 * 60 * 60, cast=int)  # 4 weeks
 SESSION_COOKIE_SECURE = config('SESSION_COOKIE_SECURE', default=not DEBUG, cast=bool)
 SESSION_COOKIE_HTTPONLY = True
-SESSION_EXPIRE_AT_BROWSER_CLOSE = False
 SESSION_COOKIE_NAME = 'session_id'
 SESSION_ENGINE = config('SESSION_ENGINE', default='django.contrib.sessions.backends.cache')
-SESSION_EXISTS_COOKIE = 'sumo_session'
```
Comment (LeoMcA) [inline kitsune/settings.py]: There's two more references to `SESSION_EXISTS_COOKIE` which need to be removed:

kitsune/users/middleware.py L24
kitsune/users/views.py L84

File: kitsune/sumo/jinja2/handlers/403.html [hunk 3]
```diff
@@ -1,27 +1,17 @@
...
       {% if not user.is_authenticated() %}
-          <p>
-            Please try to log in / sign up first with <a href="{{ url('users.fxa_authentication_init') }}">{{ _('Firefox Accounts') }}</a>
-            or continue with an existing <a class="legacy-login-toggle" href="#">{{ _('SUMO account') }}</a>.
-          </p>
-          <div id="login-legacy" class="hidden">
-            {{ login_form(form) }}
-          </div>
+        <p>
+          Please try to log in / sign up first with <a href="{{ url('users.fxa_authentication_init') }}">{{ _('Firefox Accounts') }}</a>.
```
Comment (LeoMcA) [inline kitsune/sumo/jinja2/handlers/403.html]: Should this be localised?

File: kitsune/sumo/static/sumo/js/legacy_login_toggle.js [hunk 4]
```diff
@@ -1,13 +0,0 @@
-(function() {
```
Comment (LeoMcA) [inline kitsune/sumo/static/sumo/js/legacy_login_toggle.js]: This file is still referenced in styleguide/kss-config.json

File: kitsune/bundles.py [hunk 5]
```diff
@@ -124,21 +123,6 @@
         "source_filenames": ("sumo/js/questions.metrics-dashboard.js",),
         "output_filename": "build/questions.metrics-min.js",
     },
-    "questions.aaq.react": {
```
Comment (LeoMcA) [inline kitsune/bundles.py]: We've still got a reference to this in kitsune/questions/jinja2/questions/new_question_react.html

This is rendered in kitsune/questions/views.py#aaq_react, which is rendered from #aaq if the new_aaq waffle flag is set.

I guess we want to tear all of that out too? Including the tests in kitsune/questions/tests/test_views.py:ReactAAQTests

File: kitsune/sumo/static/sumo/js/legacy_login_toggle.js (hunk 4 above)
Comment (akatsoulas) [reply kitsune/sumo/static/sumo/js/legacy_login_toggle.js]: Nice catch. I will remove it from there too.

File: kitsune/users/forms.py [hunk 6]
```diff
@@ -88,54 +75,6 @@ def save_for_user(self, user):
                 user.settings.create(name=field, value=value)
 
 
-class AuthenticationForm(auth_forms.AuthenticationForm):
```
Comment (LeoMcA) [inline kitsune/users/forms.py]: There's a reference to this form in ModelBackendAllowInactive:
```
Standard model authentication that also allows inactive users.

    This is necessary as new registered users are marked inactive
    but still logged in automatically. Some control around logging
    in as an active user is still retained via the ``only_active``
    argument of ``kitsune.users.forms.AuthenticationForm``.
```
Is it still true that new registered users are marked inactive? This might be something we can tear out later.

File: kitsune/bundles.py (hunk 5 above)
Comment (LeoMcA) [reply kitsune/bundles.py]: Do we also want a migration to remove the `new_aaq` flag from the db?

kitsune/questions/migrations/0004_new_aaq_waffle_flag.py is where its added

File: kitsune/users/management/commands/clear_expired_auth_tokens.py [hunk 7]
```diff
@@ -1,11 +0,0 @@
-from datetime import datetime, timedelta
```
Comment (LeoMcA) [inline kitsune/users/management/commands/clear_expired_auth_tokens.py]: We still have a reference to this in scripts/cron.py#job_clear_expired_auth_tokens

We should remove the DMS_CLEAR_EXPIRED_AUTH_TOKENS setting from settings.py and our k8s files, and remember to remove it from DMS when we deploy to prod.

Do we also need a migration to nuke all Token objects from the DB?

File: kitsune/users/management/commands/purge_hashes.py [hunk 8]
```diff
@@ -1,19 +0,0 @@
-from datetime import datetime, timedelta
```
Comment (LeoMcA) [inline kitsune/users/management/commands/purge_hashes.py]: Similarly, in scripts/cron.py#job_purge_hashes, same comment as above wrt DMS.

I haven't gotten there yet, but do we have a migration to remove the user.password column from the DB?

File: kitsune/users/management/commands/remove_expired_registration_profiles.py [hunk 9]
```diff
@@ -1,12 +0,0 @@
-from django.core.management.base import BaseCommand
```
Comment (LeoMcA) [inline kitsune/users/management/commands/remove_expired_registration_profiles.py]: scripts/cron.py#job_remove_expired_registration_profiles

But do we want to run this once after we deploy this?

File: kitsune/users/static/users/js/actions/UserAuthActions.es6.js [hunk 10]
```diff
@@ -1,149 +0,0 @@
-/* globals $:false */
```
Comment (LeoMcA) [inline kitsune/users/static/users/js/actions/UserAuthActions.es6.js]: This file is referenced in kitsune/questions/static/questions/js/components/UserAuth.jsx
which is referenced in kitsune/questions/static/questions/js/components/AAQApp.jsx
which is referenced in kitsune/questions/static/questions/js/aaq.browserify.js
which doesn't seem to be referenced anywhere

There's a lot of AAQ react stuff to tear out... I wonder if it should be done as a separate PR.

File: kitsune/users/static/users/js/components/LoginForm.jsx [hunk 11]
```diff
@@ -1,64 +0,0 @@
-/* globals React:false */
```
Comment (LeoMcA) [inline kitsune/users/static/users/js/components/LoginForm.jsx]: Also referenced in UserAuth.jsx

File: kitsune/users/static/users/js/components/RegisterForm.jsx [hunk 12]
```diff
@@ -1,94 +0,0 @@
-/* globals React:false */
```
Comment (LeoMcA) [inline kitsune/users/static/users/js/components/RegisterForm.jsx]: Also referenced in UserAuth.jsx

File: kitsune/users/static/users/js/constants/UserAuthConstants.es6.js [hunk 13]
```diff
@@ -1,27 +0,0 @@
-import constantMap from '../../../sumo/js/utils/constantMap.es6.js';
```
Comment (LeoMcA) [inline kitsune/users/static/users/js/constants/UserAuthConstants.es6.js]: Referenced in kitsune/questions/static/questions/js/components/UserAuth.jsx

and kitsune/questions/static/questions/js/components/SubmitQuestion.jsx

and kitsune/questions/static/questions/js/components/AttachmentField.jsx

File: kitsune/users/static/users/js/stores/UserAuthStore.es6.js [hunk 14]
```diff
@@ -1,126 +0,0 @@
-/* globals _:false */
```
Comment (LeoMcA) [inline kitsune/users/static/users/js/stores/UserAuthStore.es6.js]: Referenced in kitsune/questions/static/questions/js/actions/AAQActions.es6.js
which is referenced in... lots of places... I think all the react has to go.

File: kitsune/users/forms.py (hunk 6 above)
Comment (akatsoulas) [reply kitsune/users/forms.py]: New users are not marked inactive. Active/Inacative users is one of the things this PR does not solve. I will have a look at that reference though- Thanks

File: kitsune/users/models.py [hunk 15]
```diff
@@ -366,229 +356,6 @@ def get_for_user(cls, user, name):
...
-    log.info(log_msg.format(num=len(data)))
-    qs = RegistrationProfile.objects.filter(id__in=data)
-    for profile in qs.select_related('user'):
-        user = profile.user
-        profile.delete()
-        if user and not user.is_active:
-            user.delete()
-
-
-class EmailChangeManager(ConfirmationManager):
```
[the remark this answers is not in the document]
Comment (akatsoulas) [reply kitsune/users/models.py]: It is but I left the models intact until we decide what we will do with the underlying data in the db.

File: kitsune/users/views.py [hunk 16]
```diff
@@ -502,221 +288,6 @@ def make_contributor(request):
...
-    else:  # request.method == 'GET'
-        form = AvatarForm(instance=user_profile)
-
-    return render(request, 'users/edit_avatar.html', {
-        'form': form, 'profile': user_profile})
-
-
-@login_required
-@require_http_methods(['GET', 'POST'])
-def delete_avatar(request):
```
Comment (LeoMcA) [inline kitsune/users/views.py]: Do we allow users to unset their fxa avatar? There might be privacy implications if we don't

File: kitsune/bundles.py (hunk 5 above)
Comment (akatsoulas) [reply kitsune/bundles.py]: I would go with no at this point. We can clean up any related artifacts after the release

File: kitsune/users/management/commands/clear_expired_auth_tokens.py (hunk 7 above)
Comment (akatsoulas) [reply kitsune/users/management/commands/clear_expired_auth_tokens.py]: Probably we do but I wouldn't do it at this point. We are closing down all the paths to use these tokens. After the release we can clean everything up

File: kitsune/users/management/commands/purge_hashes.py (hunk 8 above)
Comment (akatsoulas) [reply kitsune/users/management/commands/purge_hashes.py]: It's a no about the migration.

File: kitsune/users/forms.py (hunk 6 above)
Comment (LeoMcA) [reply kitsune/users/forms.py]: I think that means we can remove `ModelBackendAllowInactive` then.

File: kitsune/questions/events.py [hunk 17]
```diff
@@ -84,10 +82,7 @@ def _make_mail(locale, user, context):
             from kitsune.users.templatetags.jinja_helpers import display_name
 
             is_asker = asker_id == user.id
-            extra_params = {}
             if is_asker:
-                auth_str = get_auth_str(user)
```
Comment (safwanrahman) [inline kitsune/questions/events.py]: This is a important feature actually. I do not know if the feature should be removed.

The user should be logged in automatically when he clicks the link. I do not know if there are any issue to remove this feature.

File: kitsune/users/views.py (hunk 16 above)
Comment (akatsoulas) [reply kitsune/users/views.py]: The avatars are updated/handled in the FxA account.

File: kitsune/questions/events.py (hunk 17 above)
Comment (akatsoulas) [reply kitsune/questions/events.py]: This will be tuned a bit to allow users to be redirected back after logging in if the don't have an active session. I was reluctant too to remove this but at the same time there is a security concern on how this is handled. Notification will still be there but users will have to login if they are not already logged in. Thoughts?

File: kitsune/settings.py [hunk 18]
```diff
@@ -513,18 +510,12 @@ def parse_conn_max_age(value):
...
-    # automatically logged. It also has to come after
-    # NoVarySessionMiddleware.
+    # This has to come after NoVarySessionMiddleware.
     'django.contrib.messages.middleware.MessageMiddleware',
 
     # This should come after MessageMiddleware
     'kitsune.sumo.middleware.SUMORefreshIDTokenAdminMiddleware',
 
-    # This middleware should come after AuthenticationMiddleware.
-    'kitsune.users.middleware.TokenLoginMiddleware',
```
Comment (safwanrahman) [inline kitsune/settings.py]: If the login with token feature is not removed, we should keep this middleware

File: kitsune/settings.py [hunk 19]
```diff
@@ -806,13 +796,10 @@ def JINJA_CONFIG():
 
 #
 # Sessions
-SESSION_COOKIE_AGE = config('SESSION_COOKIE_AGE', default=4 * 7 * 24 * 60 * 60, cast=int)  # 4 weeks
```
Comment (safwanrahman) [inline kitsune/settings.py]: What is the reason to remove it? It is a django settings

File: kitsune/users/api.py [hunk 20]
```diff
@@ -234,25 +204,6 @@ def validate(self, data):
...
-
-    def update(self, instance, validated_data):
-        if "user" in validated_data:
-            user_data = validated_data.pop("user")
-            for key, val in user_data.items():
-                setattr(instance.user, key, val)
-            instance.user.save()
-        return super(ProfileSerializer, self).update(instance, validated_data)
-
     def validate_username(self, username):
```
Comment (safwanrahman) [inline kitsune/users/api.py]: As this serializer should not be used for creating profile anymore, you should remove the validation as well.

File: kitsune/users/api.py [hunk 21]
```diff
@@ -288,10 +239,8 @@ class Meta(ProfileSerializer.Meta):
 
 
 class ProfileViewSet(
-    mixins.CreateModelMixin,
     mixins.RetrieveModelMixin,
     mixins.ListModelMixin,
-    mixins.UpdateModelMixin,
```
Comment (safwanrahman) [inline kitsune/users/api.py]: Should we remove the profile update API as well?

Comment (akatsoulas) [reply kitsune/users/api.py]: I am not sure I understand your question.

File: kitsune/users/models.py [hunk 22]
```diff
@@ -366,229 +356,6 @@ def get_for_user(cls, user, name):
...
-                        'activate_url': url,
-                        'login_url': reverse('users.login'),
-                        'reg': 'main'}
-        email_kwargs.update(kwargs)
-
-        # RegistrationProfile doesn't have a locale attribute. So if
-        # we get one of those, then we have to get the real profile
-        # from the user.
-        if hasattr(confirmation_profile, 'locale'):
-            locale = confirmation_profile.locale
```
[the remark this answers is not in the document]
Comment (LeoMcA) [reply kitsune/users/models.py]: As we discussed, we're leaving RegistrationProfile in for now until we decide what to do with the data in db

File: kitsune/questions/static/questions/js/actions/AAQActions.es6.js [hunk 23]
```diff
@@ -1,291 +0,0 @@
-/* globals _:false */
-import apiFetch from '../../../sumo/js/utils/apiFetch.es6.js';
```
Comment (LeoMcA) [inline kitsune/questions/static/questions/js/actions/AAQActions.es6.js]: apiFetch doesn't seem to be imported anywhere else, it can be removed

File: kitsune/questions/static/questions/js/actions/AAQActions.es6.js [hunk 24]
```diff
@@ -1,291 +0,0 @@
...
-import Dispatcher from '../../../sumo/js/Dispatcher.es6.js';
```
Comment (LeoMcA) [inline kitsune/questions/static/questions/js/actions/AAQActions.es6.js]: Dispatcher is only imported in UrlActions and UrlStore, neither of which are imported anywhere else, they can all be removed

File: kitsune/questions/static/questions/js/components/AAQApp.jsx [hunk 25]
```diff
@@ -1,87 +0,0 @@
...
-import UserAuthStore from '../../../users/js/stores/UserAuthStore.es6.js';
-import ProductSelector from './ProductSelector.jsx';
-import TopicSelector from './TopicSelector.jsx';
-import TitleContentEditor from './TitleContentEditor.jsx';
-import AttachmentField from './AttachmentField.jsx';
-import UserAuth from './UserAuth.jsx';
-import SubmitQuestion from './SubmitQuestion.jsx';
-import TroubleshootingDataStore from '../stores/TroubleshootingDataStore.es6.js';
-import UrlStore from '../../../sumo/js/stores/UrlStore.es6.js';
-import {pathStructure} from '../../../sumo/js/constants/UrlConstants.es6.js';
```
Comment (LeoMcA) [inline kitsune/questions/static/questions/js/components/AAQApp.jsx]: UrlConstants is only imported in UrlActions and UrlStore neither of which are imported anywhere else, they can all be removed

File: kitsune/questions/static/questions/js/components/AAQStep.jsx [hunk 26]
```diff
@@ -1,49 +0,0 @@
-/* globals React:false */
-import AAQActions from '../actions/AAQActions.es6.js';
-import scrollTo from '../../../sumo/js/utils/scrollTo.es6.js';
```
Comment (LeoMcA) [inline kitsune/questions/static/questions/js/components/AAQStep.jsx]: scrollTo doesn't seem to be imported anywhere else, it can be removed

File: kitsune/questions/static/questions/js/constants/AAQConstants.es6.js [hunk 27]
```diff
@@ -1,28 +0,0 @@
-import constantMap from '../../../sumo/js/utils/constantMap.es6.js';
```
Comment (LeoMcA) [inline kitsune/questions/static/questions/js/constants/AAQConstants.es6.js]: constantMap is only imported in UrlConstants which is only imported in UrlActions and UrlStore, they can all be removed

File: kitsune/questions/static/questions/js/stores/QuestionEditStore.es6.js [hunk 28]
```diff
@@ -1,134 +0,0 @@
-/* globals _:false */
-
-import BaseStore from '../../../sumo/js/stores/BaseStore.es6.js';
```
Comment (LeoMcA) [inline kitsune/questions/static/questions/js/stores/QuestionEditStore.es6.js]: BaseStore is only imported in UrlStore, which isn't imported anywhere else, they can all be removed

File: kitsune/questions/static/questions/js/utils/aaqGa.es6.js [hunk 29]
```diff
@@ -1,4 +0,0 @@
-import googleAnalytics from '../../../sumo/js/utils/googleAnalytics.es6.js';
```
Comment (LeoMcA) [inline kitsune/questions/static/questions/js/utils/aaqGa.es6.js]: this is the only reference to googleanalytics.es6.js, it can be removed

**Comment by akatsoulas** [comment]
After discussing with @LeoMcA in slack, we will clean up any react left overs after the release

File: kitsune/settings.py (hunk 19 above)
Comment (akatsoulas) [reply kitsune/settings.py]: This has been restored. Probably this is not marked as outdated because I moved around a few commits in order to make them more atomic. Nice catch

**Comment by akatsoulas** [comment]
After talking with @safwanrahman we will continue reviewing and testing this PR after the merge. For now this is merged in order to get some QA help on it.
